reference.md

```diff
@@ -499,18 +542,154 @@ variables are missing:
 
 ``` fennel
 ;; bad
-(match [1 2 3]
+(case [1 2 3]
   ;; Will throw an error because `b' is nil for the first
   ;; pattern but the guard still uses it.
   (where (or [a 1 2] [a b 3]) (< a 0) (< b 1))
   :body)
 
 ;; ok
-(match [1 2 3]
+(case [1 2 3]
   (where (or [a b 2] [a b 3]) (< a 0) (<= b 1))
   :body)
 ```
 
+#### Binding Unification
+
+Symbols bound inside a `case` pattern are independent from any existing
+symbols in the current scope, that is - names may be re-used without
+consequence.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; `x` is simply bound to the first value of [:hello]
+    [x] x)) ; -> :hello
+```
+
+Sometimes it may be desirable to match against an existing value in the outer
+scope. To do this we can "unify" a binding inside the pattern with an existing
+outer binding with the unary `(= my-symbol)` form. The unary `(= my-symbol)`
+form is *only* valid in a `case` pattern.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; 1 != :hello
+    [(= x)] x
+    _ :no-match)) ; -> no-match
+
+(let [x 1]
+  (case [1]
+    ;; 1 == 1
+    [(= x)] x
+    _ :no-match)) ; -> 1
+
+(let [pass :hunter2]
+  (case (user-input)
+    (= pass) :login
+    _ :try-again!))
+```
+
+Unification is only required inside the pattern. Outer bindings are
+automatically available inside guards and bodies as long as the name has not
+been rebound in the pattern.
+
+Example:
+
+```fennel
+(let [limit 10]
+  (case [9 9]
+    ;; automatically avaliable in the guard
```

Comment:
I'm not sure this is helpful; it seems to just be pointing out that "scoping and variables work like normal here", right? Shouldn't that be the default assumption?

Reply:
Someone new to pattern matching, the term "bindings", "pinning", etc might wonder the rules so I don't think it hurts to reinforce them. It wont hurt those that know but will help those that wonder.

Reply:
I think this is distracting; this section should focus on how `case` works and not reviewing how scoping works in general. It also seems to imply that there might be somewhere else where scoping doesn't work this way, which is a bit misleading.

Other than that, I think this is ready to go; thanks for that!

Reply:
Perhaps if the pinning code example above added another local in the outer `let` which one of the bodies used, this could be demonstrated more subtly in a way that doesn't call extra attention to it?